### ISSUE TITLE
Add distinct for entity manager find options

### ISSUE THREAD

**Issue by ziofat** [typeorm/typeorm#5163]
**Issue type:**

[ ] question
[ ] bug report
[x] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

It will be nice to have distinct option for select in find options when using entity manager.

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `mysql`
    [ ] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
!>

**Comment by swim** [comment]
Agreed, exposing the distinct option to the find method would be super handy. However, there's a minor complication to this request. The find method maps to getMany which in turn sets the expression map query entity property to true. This changes the SQL statement by including the id in the selection which breaks the distinct result set.
